conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/AuthTest.java

```diff
@@ -55,7 +64,35 @@ public void testAuthHeaderMultipleValues() {
         exchange.getRequestHeaders().add(Headers.AUTHORIZATION, "Bearer foo");
         exchange.getRequestHeaders().add(Headers.AUTHORIZATION, "Bearer bar");
         assertThatThrownBy(() -> CONTEXT.auth().header(exchange))
-                .isInstanceOf(SafeIllegalArgumentException.class)
-                .hasMessage("One Authorization header value is required");
+                .isInstanceOf(ServiceException.class)
+                .hasMessageContaining("ServiceException: UNAUTHORIZED (Conjure:MalformedCredential)");
+    }
+
+    //delete
+    @Test
+    public void testParseAuthCookie() {
+        BearerToken expected = BearerToken.valueOf("token");
+        String cookieName = "Auth-Token";
+        HttpServerExchange exchange = HttpServerExchanges.createStub();
+        exchange.getRequestCookies().put(cookieName, new CookieImpl(cookieName, "token"));
+        assertThat(CONTEXT.auth().cookie(exchange, cookieName)).isEqualTo(expected);
+    }
+
+    @Test
+    public void testAuthCookieNotPresent() {
+        HttpServerExchange exchange = HttpServerExchanges.createStub();
+        assertThatThrownBy(() -> CONTEXT.auth().cookie(exchange, "any"))
+                .isInstanceOf(ServiceException.class)
+                .hasMessage("ServiceException: UNAUTHORIZED (Conjure:MissingCredential)");
```

Comment:
we should probably use the assertj matchers for these guys rather than assert things on strings.
https://github.com/palantir/conjure-java-runtime-api/blob/develop/test-utils/src/main/java/com/palantir/conjure/java/api/testing/ServiceExceptionAssert.java